### DISCUSSION TITLE
How to use .md file in doc_md for DAG documentation

### DISCUSSION THREAD

**Discussion by a-kumar5** [apache/airflow#42014] [General] [answered]
### What do you see as an issue?

![image](https://github.com/user-attachments/assets/0e763c1c-4f3f-402e-bab1-eb2c26b40315)

### Solving the problem

Please add the steps to showcase how the .md file can be used for dag documentation under doc_md. 

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by jscheffl** [comment]
Like here? https://github.com/apache/airflow/blob/main/airflow/example_dags/example_params_ui_tutorial.py#L38

**Comment by jscheffl** [comment]
Else have you seen/taken a look at https://github.com/apache/airflow/blob/main/airflow/models/dag.py#L706 ?

**Comment by a-kumar5** [comment]
@jscheffl  hers is my dag code.
![image](https://github.com/user-attachments/assets/f522f860-85f7-4efe-a920-b84886d0919e)

I’m currently facing an issue where a templating error is being thrown, stating:

"Templating Error! Not able to find the template file: /opt/airflow/dags/TESTING_1/docs/00AKOSB3-FLSBH_3.md."

However, I have verified by exec-ing into the pod that the file does indeed exist at the specified location.

Could you please provide guidance on what might be causing this error or suggest any steps I could take to troubleshoot this further?

![image](https://github.com/user-attachments/assets/58a347f9-68a4-418d-9daa-b751c815b242)

**Comment by a-kumar5** [comment]
![image](https://github.com/user-attachments/assets/d407d7ab-ce31-4af7-bb9e-f281a4b5297c)

**Comment by a-kumar5** [comment]
@jscheffl  Additionally, I was able to print the contents of the .md file from within the task using the same path, which suggests that the file is accessible. 

![image](https://github.com/user-attachments/assets/5d9a9549-9d4c-44b5-b019-190eb5254771)

![image](https://github.com/user-attachments/assets/0b248f5e-4fc5-4e2a-bfc9-e71035721e4d)

**Comment by jscheffl** [comment]
I checked with latest main which seems not to have changed since Airflow 2.9.3 and I can tell that the function works. Just with the side note that if a relative path is given then the file is _not_ loaded from DAGs folder but from working directory in which the Airflow Scheduler component was started.

Note that the MD file content is parsed by the scheduler. This means if you save the file, it not immeditately displayed but only after some time when the Scheduler component did a re-parse of the DAG file.

As the screenshots you have posted are a bit "older" - can you reveal which Airflow-Version you are running? The feature was introduced in Airflow 2.4

**Comment by a-kumar5** [comment]
@jscheffl 
Currently running 2.8.3v

![image](https://github.com/user-attachments/assets/1b403d05-8471-40c7-a094-37f94463ec9d)

**Comment by a-kumar5** [comment]
@jscheffl I provided the absolute path `/opt/airflow/dags/TESTING_1/docs/00AKOSB3-FLSBH_3.md`. This approach works perfectly in my local environment, where I installed Airflow using Docker. However, in the EKS cluster, it's throwing a `template not found` error.

In my local environment, where I'm using Docker, Airflow is running on version `2.9.3`, while in the EKS cluster, it's running on version `2.8.3`.

I last edited the DAG code 24 hours ago, yet it’s still throwing a `template not found` error. Has the scheduler not parsed it again in the last 24 hours, or am I missing something?

**Comment by jscheffl** [comment, marked as answer]
Okay, you hit a special situation as the logic was adjusted between 2.8.3 and 2.9.3 because of a security hardening. Templating support was removed in 2.9.3.

So I propose that you test with the same version as well I propose to upgrade to the higher if possible. THen you have the same behavior.

24h should be sufficient but be aware that the the context path of the scheduler in EKS will be probably different than in your local setup.
As I found a bit of inconsistency with the changes in 2.9.3 and the docs, please see that I adjusted the docs. Please do not resolve the MD file path relative to the DAG file path but relative to the path in which the scheduler was started via `os.getcwd()`.

**Comment by jscheffl** [comment]
As this is more a link of discussion/troubleshooting, converting to a discussion.
